### PR TITLE
SOFA conversion for TAU-SRIR database (and others)

### PR DESCRIPTION
**Pull request by ChrisIck**
Includes the `sofa_utils.py` module with some basic functions for storing a SOFA conventions-compliant file from an array of SRIRs, and a set of corresponding measurement locations. This is following the room SRIR conventions established in the ARNI dataset, which was also released in SOFA format.

Because this is dependent on open PR #18 , I'm closing that PR and also including the room simulation code here, including a demo notebook on how it works, the module `room_sim.py` with all of the relevant ISM simulation code, as well as some utilities in `tau_loading.py` for dealing with TAU-SRIR-DB structure.

### CODE REVIEW

File: room_simulation/room_sim.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+                if not os.path.exists(path_filename):
+                    with h5py.File(path_filename, 'w') as f:
+                        f.create_dataset('rirs', data=path_rirs)
+                        f.create_dataset('rate', data=np.array([args.sr]))
+                else:
+                    with h5py.File(path_filename, 'r+') as f:
+                        f['rirs'][:] = path_rirs
+                        f['rate'][:] = np.array([args.sr])
+                print("Stored path at {}".format(path_filename))
+
```
Comment (ChrisIck) [inline room_simulation/room_sim.py:301]: TODO: Switching from pickle files to SOFA databases

**Comment by ChrisIck** [comment]
@iranroman Added flipping, ready for commit

File: data/convert_tau_sofa.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import sofa_utils
+import os
+
+tau_db_dir = '/scratch/ci411/SRIR_DATASETS/TAU_SRIR_DB/TAU-SRIR_DB'
+sofa_db_dir = '/scratch/ci411/SRIR_DATASETS/TAU_SRIR_DB_SOFA'
+db_name = "TAU-SRIR-DB-SOFA"
```
Comment (iranroman) [inline data/convert_tau_sofa.py:6]: When I first ran this I got an error:
  
  `PermissionError: [Errno 13] Permission denied: 'TAU_DB/TAU_SRIR_DB_SOFA/foa/bomb_shelter.sofa`
  
  which was solved by `mkdir -p TAU_DB/TAU_SRIR_DB_SOFA/foa`. Let's include this step as python code in this preprocessing step.

**Comment by iranroman** [comment]
Hello Chris, I moved things around a little bit, but the code works very well as you provided it. I only got a minor error due to the implicit step of creating the directories for the sofa database beforehand. Could you address that? `convert_tau_sofa.py` is now under `scripts`

File: data/sofa_demo.ipynb [hunk 3]
```diff
@@ -0,0 +1,75 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sofa_utils\n",
```
Comment (iranroman) [inline data/sofa_demo.ipynb:9]: update import to be `from room_scaper import sofa_utils`

File: data/sofa_demo.ipynb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "load_rirs = sofa_utils.load_rir('sofa_file.sofa')\n",
+    "load_sp = sofa_utils.load_pos('sofa_file.sofa')"
```
Comment (iranroman) [inline data/sofa_demo.ipynb:41]: this loads the absolute coordinates, right? How can I get them normalized?

**Comment by iranroman** [comment]
I left a couple more questions/observations. As soon as those are addressed, I will try to generate data. Also, it would be nice to have the room "trajectories" in separate files.

File: data/sofa_demo.ipynb (hunk 4 above)
Comment (ChrisIck) [reply data/sofa_demo.ipynb:41]: [See here](https://github.com/iranroman/SELD-data-generator/pull/20/files#diff-1d64364bdf0922904a7de475e2e66ba59e16168e2741badd29bc90268e09f138R170), theres the kwarg `doas` which is `True` by default; you're already loading normalized coordinates! If you set `doas` to false, you'll get the aboslute coordinates.

File: data/convert_tau_sofa.py (hunk 2 above)
Comment (ChrisIck) [reply data/convert_tau_sofa.py]: Added in commit `[a08e28c](https://github.com/iranroman/SELD-data-generator/pull/20/commits/a08e28c023571c31312c27f2946e698e06585d88)`

**Comment by iranroman** [comment]
@sivannavis has validated model performance with the mixtures generated by this PR. We are ready to merge.

**Pull request merged**